### PR TITLE
[jspi] - Use new -sJSPI flag to enable JSPI.

### PR DESCRIPTION
**Pull request by brendandahl**

### CODE REVIEW

File: src/settings.js [hunk 1]
```diff
@@ -805,14 +805,14 @@ var NODEJS_CATCH_REJECTION = true;
...
+//   but it has a significant cost in terms of code size and speed.
 //   See https://emscripten.org/docs/porting/asyncify.html
-// - Depend on VM support for the wasm stack switching proposal. This allows
-//   async operations to happen without the overhead of modifying the wasm.
-//   This is experimental atm while spec discussion is ongoing, see
-//   https://github.com/WebAssembly/js-promise-integration/
+// - 'jspi': Depend on VM support for the JavaScript Promise Integration
+//   proposal. This allows async operations to happen without the overhead of
+//   modifying the wasm. This is experimental atm while spec discussion is
+//   ongoing, see https://github.com/WebAssembly/js-promise-integration/
```
Comment (sbc100) [inline src/settings.js:815]: Should we mention which on is the default if you just so `-sASYNCIFY` (I assume its binaryen)?

**Comment by dschuff** [comment]
I forget whether we discussed this before, but have we considered just using `-sJSPI` instead of `-sASYNCIFY=jspi`? This would seem to make sense in the long term where JSPI is supported everywhere and there's really no reason to use binaryen-based asyncify.

**Comment by sbc100** [comment]
> I forget whether we discussed this before, but have we considered just using `-sJSPI` instead of `-sASYNCIFY=jspi`? This would seem to make sense in the long term where JSPI is supported everywhere and there's really no reason to use binaryen-based asyncify.

I think we did discuss that option yes.    I don't feel particularly strongly either way.

**Comment by brendandahl** [comment]
We discussed a bit, but didn't really come to a firm decision. Binaryen's asyncify and JSPI both use a lot of the same "asyncify" code in emscripten, so I think internally for emscripten development it makes sense to keep these under the same flag.

If the long term goal is to remove binaryen asyncify, having a separate `-sJSPI` setting makes more sense.

**Comment by dschuff** [comment]
I think in the long term, all browsers will support JSPI, and in that world I don't think there would be any reason that emscripten users would want to use Binaryen asyncify. (Do y'all agree?) If so, that might argue for defining a "JSPI" flag which could just alias internally to ASYNCIFY=2, and then somewhere down the road (though probably not very soon) we could remove the support at least from emscripten.

**Comment by dschuff** [comment]
It's probably also worth putting a note in the changelog that 'JSPI' is now supported and the recommended flag.

**Comment by brendandahl** [comment]
We're probably also going to switch to JSPI_IMPORTS and JSPI_EXPORTS soon too. That will be nice to highlight which settings are supported/not supported for JSPI vs ASYNCIFY.

**Review by sbc100 — APPROVED** [review]

**Comment by dschuff** [comment]
> We're probably also going to switch to JSPI_IMPORTS and JSPI_EXPORTS soon too. That will be nice to highlight which settings are supported/not supported for JSPI vs ASYNCIFY.

Good point. I'm fine with waiting on the changelog entry until then, I expect it won't be long anyway.

**Comment by curiousdannii** [comment]
> It's probably also worth putting a note in the changelog that 'JSPI' is now supported and the recommended flag.

JSPI shouldn't be the recommended option until #21081 is resolved.

**Pull request merged**